src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -2,20 +2,72 @@ import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { Exam } from 'app/entities/exam.model';
 import { ExamManagementService } from 'app/exam/manage/exam-management.service';
+import { Observable } from 'rxjs';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { AlertService } from 'app/core/alert/alert.service';
+import { Course } from 'app/entities/course.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
 
 @Component({
     selector: 'jhi-exam-update',
     templateUrl: './exam-update.component.html',
 })
 export class ExamUpdateComponent implements OnInit {
     exam: Exam;
+    course: Course;
+    isSaving: boolean;
 
-    constructor(private route: ActivatedRoute, private examManagementService: ExamManagementService) {}
+    constructor(
+        private route: ActivatedRoute,
+        private examManagementService: ExamManagementService,
+        private jhiAlertService: AlertService,
+        private courseManagementService: CourseManagementService,
+    ) {}
 
-    /**
-     * Initialize the exam
-     */
     ngOnInit(): void {
-        this.route.data.subscribe(({ exam }) => (this.exam = exam));
+        this.route.data.subscribe(({ exam }) => {
+            this.exam = exam;
+            this.courseManagementService.find(Number(this.route.snapshot.paramMap.get('courseId'))).subscribe(
+                (response: HttpResponse<Course>) => {
+                    this.exam.course = response.body!;
+                    this.course = response.body!;
+                },
+                (err: HttpErrorResponse) => this.onError(err),
+            );
```

Comment:
You could also use `CourseResole` here. Just add it in your route in `exam-management.route`.

Reply:
Discussed it with @sascha11110 and decided not to use it